cmake/clang-format.cmake

```diff
@@ -0,0 +1,31 @@
+# select all files with correct c++ extensions
+file(GLOB_RECURSE
+  ALL_CXX_SOURCE_FILES
+  *.[ch] *.[chi]pp *.[chi]xx *.cppm *.cc *.hh *.ii *.[CHI]
+)
+
+# exclude the docs **for now**
+set(EXCLUDE_REGEX ".*doc[\\|/].*")
+if(${CMAKE_VERSION} VERSION_LESS "3.6.0")
```

Comment:
No need to "dereference" CMake variables in `if` conditions. Simply write `if(CMAKE_VERSION VERSION_LESS "3.6.0")`. Same for the `if` below on line 22.

Reply:
Nice. Didn't know that. Will modify the `if` statements

Reply:
Didn't work for `MATCHES`, but resolved in the 2 cases you mentioned.